### PR TITLE
[DOC] Small fix for Histogram documentation.

### PR DESCRIPTION
**Pull request by lalitb**
Fixes # (issue)

## Changes

The comment is not consistent with [implementation ](https://github.com/open-telemetry/opentelemetry-cpp/blob/61e8741a70cde8f273b3de49a772f05a12566e80/sdk/src/metrics/sync_instruments.cc#L185 )and [specification](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#record). The Histogram API only records non-negative value.

Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Unit tests have been added
* [x] Changes in public API reviewed

### CODE REVIEW

File: api/include/opentelemetry/metrics/sync_instruments.h [hunk 1]
```diff
@@ -97,14 +97,14 @@ class Histogram : public SynchronousInstrument
   /**
    * Records a value.
    *
-   * @param value The increment amount. May be positive, negative or zero.
+   * @param value The increment amount. MUST be non-negative.
```
Comment (marcalff) [inline api/include/opentelemetry/metrics/sync_instruments.h:100]: The comment also says "The increment amount", which sounds like a copy and paste from counters, and does not apply to histograms.

In my understanding:

For an histogram, the metric value should be passed, and will decide which bucket stat to increment.

There is no such thing as giving an "increment amount".

Comment (lalitb) [reply api/include/opentelemetry/metrics/sync_instruments.h]: Good point. Have updated this now.

**Review by marcalff — APPROVED** [review]
LGTM, thanks for the fix.

**Review by esigo — APPROVED** [review]
LGTM
Thanks :)

**Pull request merged**